src/wasm2asm.h

```diff
@@ -190,13 +191,137 @@ class Wasm2AsmBuilder {
   void addImport(Ref ast, Import *import);
   void addTables(Ref ast, Module *wasm);
   void addExports(Ref ast, Module *wasm);
+  void addWasmCompatibilityFuncs(Module *wasm);
 
   Wasm2AsmBuilder() = delete;
   Wasm2AsmBuilder(const Wasm2AsmBuilder &) = delete;
   Wasm2AsmBuilder &operator=(const Wasm2AsmBuilder &) = delete;
 };
 
+static Function* makeCtzFunc(MixedArena& allocator, UnaryOp op) {
+  assert(op == CtzInt32 || op == CtzInt64);
+  Builder b(allocator);
+  // if eqz(x) then 32 else (32 - clz(x ^ (x - 1)))
+  bool is32Bit = (op == CtzInt32);
+  Name funcName = is32Bit ? Name(CTZ32) : Name(CTZ64);
+  BinaryOp subOp = is32Bit ? SubInt32 : SubInt64;
+  BinaryOp xorOp = is32Bit ? XorInt32 : XorInt64;
+  UnaryOp clzOp = is32Bit ? ClzInt32 : ClzInt64;
+  UnaryOp eqzOp = is32Bit ? EqZInt32 : EqZInt64;
+  WasmType argType = is32Bit ? i32 : i64;
+  Binary* xorExp = b.makeBinary(xorOp, b.makeGetLocal(0, i32),
```

Comment:
the convention in other code is
````
b.makeBinary(
  xorOp,
  b.makeGetLocal(0, i32),
  b.makeBinary(
    subOp,
    b.makeGetLocal(0, i32),
    b.makeConst(Literal(1))
  )
);
````
(i.e. always nest, and always 2 spaces for nesting)

also, when creating a literal, we normally use an explicit type, `Literal(int32_t(1))` so there is no ambiguity. however, i suppose there is no ambiguity for int32 constants, even on 64-bit, is that right?